### ISSUE TITLE
batch wgs: use gene annotation file to limit "access" to genic regions

### ISSUE THREAD

**Issue by etal** [etal/cnvkit#145]
To improve performance on WGS data it's best to skip the `coverage` calculation in intergenic regions. If `--annotate refFlat.txt` is provided (for gene annotation), as `--method wgs` requires, then it's also possible to extract the coordinates of all genic regions and limit the "access" BED file to those regions.

**Comment by brentp** [comment]
this is just for annotation, right? the CN calculations would remain the same?

**Comment by etal** [comment]
No, it's to make `batch` behave more like bcbio-nextgen without requiring additional work from the user -- only calculate binned coverages and perform CNV calling in genomic regions near a gene (maybe including a small flank) when `--method wgs` is used, unless the user specifies a different region for binning with `-t/--targets`. All the necessary info is in the gene annotation file, but currently, the user has to create this BED separately to avoid very long run times that cover the whole sequencing-accessible genome.

**Comment by etal** [comment]
Correction: bcbio-nextgen uses a much larger bin size outside genic regions, treating them like off-target bins. In `batch`, maybe 10x larger is reasonable, where the genic region bin size comes from `autobin`.

The option `--target-avg-size` should override this behavior.

**Comment by chapmanb** [comment]
Thanks exactly right, bcbio treats WGS with refined target bins in transcript regions and larger antitarget bins in the remainder of the genome:

https://github.com/chapmanb/bcbio-nextgen/blob/04e397f2aa9e90df6ab64fce920d571ab8c091f5/bcbio/structural/regions.py#L244
https://github.com/chapmanb/bcbio-nextgen/blob/04e397f2aa9e90df6ab64fce920d571ab8c091f5/bcbio/structural/regions.py#L72

We're mainly trying to keep down the runtime while still giving good resolution within gene regions and reasonable resolution elsewhere. Happy to hear feedback from other approaches or what you settle on in CNVkit.

**Comment by etal** [comment]
Since the 'wgs' mode is also used for low-pass WGS, it is safer for the 'batch' command to support this as either a separate mode called 'wgs-hybrid', or a new, non-default option like `--intergenic-bin-scale=10`.
